app/models/station.rb

```diff
@@ -0,0 +1,77 @@
+class Station < ActiveRecord::Base
+  validates :name,
+            :dock_count,
+            :city,
+            :installation_date, presence: true
+
+  has_many :start_trips, class_name: :Trip, foreign_key: :start_station_id
+  has_many :end_trips, class_name: :Trip, foreign_key: :end_station_id
+
+  def self.total_stations
+    count
+  end
+
+  def self.average_number_of_bikes
+    average(:dock_count).round(2)
+  end
+
+  def self.most_bikes_available
+    maximum(:dock_count)
+  end
+
+  def self.with_most_bikes
+    find_by(dock_count: most_bikes_available)
+  end
+
+  def self.fewest_bikes_available
+    minimum(:dock_count)
+  end
+
+  def self.with_fewest_bikes
+    find_by(dock_count: fewest_bikes_available)
+  end
+
+  def self.newest_station
+    find_by(installation_date: maximum(:installation_date))
+  end
+
+  def self.oldest_station
+    find_by(installation_date: minimum(:installation_date))
+  end
+
+  def rides_started
+    return "" if start_trips.empty?
+    start_trips.count
+  end
+
+  def rides_ended
+    return "" if end_trips.empty?
+    end_trips.count
+  end
+
+  def most_common_destination
+    return "" if start_trips.empty?
+    start_trips.most_popular_ending_station.name
+  end
+
+  def most_common_origination
+    return "" if end_trips.empty?
+    end_trips.most_popular_starting_station.name
```

Comment:
i'd prefer if you would pass objects around (for example, this method would return a Station object) instead of data within a specific object.

what are some advantages to only passing objects around rather than specific pieces of data?